docs/fs.md

```diff
@@ -0,0 +1,146 @@
+ZJS API for File System
+==================
+
+* [Introduction](#introduction)
+* [API Documentation](#api-documentation)
+* [Sample Apps](#sample-apps)
+
+Introduction
+------------
+ZJS provides File System API's which match Node.js' FS module. We describe them here as there could
+potentially be minor differences. It should be noted that by default the FS module only contains
+the synchronous Node.js API's. The asynchronous API's can be compiled in by enabling the pre-processor
+define `ZJS_FS_ASYNC_APIS`. They are compiled out because all of Zephyr's File System API's are
+synchronous, so making the JavaScript API's asynchronous was only adding ROM space.
+
+Web IDL
+-------
+This IDL provides an overview of the interface; see below for documentation of
+specific API functions.
+
+```javascript
+
+interface Stat {
+    boolean isFile();
+    boolean isDirectory();
+};
+```
+
+API Documentation
+-----------------
+
+### openSync
+`object openSync(string path, string mode);`
+
+Opens a file.
+
+`path` is the name/path of the file to open
+
+`mode` is the mode to open the file in (w, r, w+ etc)
```

Comment:
These are just to satisfy the Node API. On Zephyr they really don't do anything.